consensus/quorum/quorum.go

```diff
@@ -217,7 +217,35 @@ func (s *cIdentities) NthNext(pubKey *bls.PublicKeyWrapper, next int) (bool, *bl
 	return found, &s.publicKeys[idx]
 }
 
-// NthNextHmy return the Nth next pubkey of Harmony nodes, next can be negative number
+// NthNextValidatorHmy return the Nth next pubkey nodes, but from another validator.
+func (s *cIdentities) NthNextValidatorHmy(instance shardingconfig.Instance, slotList shard.SlotList, pubKey *bls.PublicKeyWrapper, next int) (bool, *bls.PublicKeyWrapper) {
```

Comment:
it would be good if you validate publicKeys here:

```golang
if len(s.publicKeys)==0 {
   return false, nil, fmt.Errorf("...")
}
```
maybe you can check it with a minimum number of publicKeys rather than 0???

Reply:
added check